src/main/java/de/tum/in/www1/artemis/service/metis/ConversationService.java

```diff
@@ -76,7 +76,7 @@ public Conversation createConversation(Long courseId, Conversation conversation)
         List<Conversation> existingConversations = conversationRepository.findConversationsOfUserWithConversationParticipants(course.getId(), user.getId());
         Optional<Conversation> existingConversation = existingConversations.stream().filter((conversation1) -> conversation1.getConversationParticipants().stream()
                 .anyMatch(conversationParticipant -> conversationParticipant.getUser().getId().equals(
-                        conversation.getConversationParticipants().toArray(new ConversationParticipant[conversation.getConversationParticipants().size()])[0].getUser().getId())))
+                        conversation.getConversationParticipants().toArray(new ConversationParticipant[0])[0].getUser().getId())))
```

Comment:
The whole toArray() call is not needed since afterwards the element at index 0 gets accessed. 
```suggestion
                        conversation.getConversationParticipants()..iterator().next().getUser().getId())))
```

Reply:
Sometimes cleaning up code shows more opportunities :D. Applied the suggestion.